### PR TITLE
saba-ja: added formatted_val for serializable and array types

### PR DESCRIPTION
**Pull request by saba-ja**
| | |
|:---|:---|
|**_Originating Project/Creator_**| F' Tools 2.0 |
|**_Affected Component_**| serializable_type, array_type |
|**_Affected Architectures(s)_**| NA |
|**_Related Issue(s)_**| https://github.com/fprime-community/fprime-gds/pull/27 |
|**_Has Unit Tests (y/n)_**| No |
|**_Builds Without Errors (y/n)_**| Yes |
|**_Unit Tests Pass (y/n)_**| No |
|**_Documentation Included (y/n)_**| No |

---
## Change Description

Format serializable members and array elements based on the given format.

## Rationale

Currently serializable and array objects are only formatted as str without formatting members or elements of the object.
This PR adds a formatted_val property to serializable and array types.
The formatted_val will iterate all members of object and format them based on the given str format.

Note 1: All elements will be cast to str
Note 2: If a member of a serializable is an array it will call array formatted_val
Note 3: If an element of an array is a serializable it will call serializable formatted_val

## Testing/Review Recommendations

Manually tested on F Prime Ref.

Before format:
Screen Shot 2021-08-18 at 4 15 23 AM

After format:
Screen Shot 2021-08-18 at 3 57 38 AM

Note: This PR should be merged before GDS [PR 27](https://github.com/fprime-community/fprime-gds/pull/27) to avoid code break.

## Future Work

Might be a good idea to find a better way to display complex tlm values in the HTML table. The raw text view is hard to follow.

### CODE REVIEW

**Review by LeStarch — DISMISSED** [review]

**Comment by saba-ja** [comment]
Not sure what is wrong with the code format. The output error here is not helpful to figure out the what is it complaining about:

Screen Shot 2021-08-19 at 10 40 06 PM

**Comment by LeStarch** [comment]
It wants the formatter to be run:

```
pip install black==21.6b0
black fprime-tools/
```

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
I'll merge as soon as we have the LGTM result.
